test/functional/e2e/user-profile.e2e-spec.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  getUser,
+  removeUserMutation,
+} from '../user-management/user.request.params';
+import puppeteer from 'puppeteer';
+import RegistrationPage from './registration-page-object';
+import { uniqueId } from '@test/utils/mutations/create-mutation';
+import UserProfilePage from './user-profile-page-object';
+import LoginPage from './login-page-object';
+
+let userId;
+const email = `mail-${uniqueId}@alkem.io`;
+const password = 'test45612%%$';
+const firstName = 'testFN';
+const lastName = 'testLN';
+const userFullName = firstName + ' ' + lastName;
+const userProfilePage = new UserProfilePage();
+const registrationPage = new RegistrationPage();
+
+const fullNameChange = 'change';
+const firstNameChange = 'change';
+const lastNameChange = 'change';
+const countryName = 'bulgaria';
+const city = 'Test City';
+const phone = '+359777777777';
+const bio = 'Test account:  Bio information';
+const skills = 'skill1';
+const keywords = `keyword1`;
+const referenceName = `TestRefName`;
+const referenceValue = `https://www.test.com`;
+
+describe('User profile update smoke tests', () => {
+  let browser: puppeteer.Browser;
+  let page: puppeteer.Page;
+  beforeAll(async () => {
+    browser = await puppeteer.launch({
+      defaultViewport: null,
+      args: ['--window-size=1920,1040'],
+    });
+  });
+
+  afterEach(async () => {
+    const client = await page.target().createCDPSession();
+    await client.send('Network.clearBrowserCookies');
+  });
+
+  afterAll(async () => {
+    const requestUserData = await getUser(email);
+    userId = requestUserData.body.data.user.id;
+    await removeUserMutation(userId);
+    await browser.close();
+  });
+
+  describe('add task to the list', () => {
+    test('User updates its profile successfully', async () => {
```

Comment:
I feel like this can be split into several smaller tests with shared code in `beforeEach` or `beforeAll`.